src/Components/Payment/PaymentForm.tsx

```diff
@@ -0,0 +1,288 @@
+                      <Flex flexDirection="column">
+                        <Serif
+                          mb={1}
+                          size="3t"
+                          color="black100"
+                          lineHeight={18}
+                        >
+                          Credit Card
+                        </Serif>
+                        <CreditCardInput
+                          error={error}
+                          onChange={response =>
+                            this.setState({ error: response.error })
+                          }
+                          ref={el => (this.cardElement = el)}
+                        />
+                      </Flex>
+
+                      <AddressForm
+                        value={this.state.address}
+                        onChange={address => this.setState({ address })}
+                        errors={this.state.addressErrors}
+                        touched={this.state.addressTouched}
+                        billing
+                      />
+                      <Button
+                        size="large"
+                        width="100%"
+                        onClick={this.onSubmit}
+                        loading={isCommittingMutation}
+                      >
+                        Submit
+                      </Button>
+                    </Join>
+                    <Spacer mb={3} />
+                  </Box>
+                </Flex>
+              )
+            }}
+          </Responsive>
+          <ErrorModal
+            onClose={this.onCloseModal}
+            show={this.state.isErrorModalOpen}
+            contactEmail="support@artsy.net"
+            detailText={this.state.errorModalMessage}
+          />
+        </>
+      </Theme>
+    )
+  }
+
+  onCloseModal = () => {
+    this.setState({ isErrorModalOpen: false })
+  }
+
+  private validateAddress(address: Address) {
+    const { name, addressLine1, city, region, country, postalCode } = address
+    const usOrCanada = country === "US" || country === "CA"
+    return {
+      name: validatePresence(name),
+      addressLine1: validatePresence(addressLine1),
+      city: validatePresence(city),
+      region: usOrCanada && validatePresence(region),
+      country: validatePresence(country),
+      postalCode: usOrCanada && validatePresence(postalCode),
+    }
+  }
+
+  private getSelectedBillingAddress(): stripe.TokenOptions {
+    const {
+      name,
+      addressLine1,
+      addressLine2,
+      city,
+      region,
+      postalCode,
+      country,
+    } = this.state.address
+
+    return {
+      name,
+      address_line1: addressLine1,
+      address_line2: addressLine2,
+      address_city: city,
+      address_state: region,
+      address_zip: postalCode,
+      address_country: country,
+    }
+  }
+
+  onCreditCardAdded(me, store, data): void {
+    const {
+      createCreditCard: { creditCardOrError },
+    } = data
+
+    if (creditCardOrError.creditCardEdge) {
```

Comment:
Can you add a small comment describing whats happening for those not familiar with relay internals?